CONTRIBUTING.md

```diff
@@ -0,0 +1,64 @@
+It also communicates that you agree to respect the time of the developers managing and developing these open source projects.
+In return, we will reciprocate that respect by addressing your issue, assessing changes, and helping you finalize your pull requests.
+
+## Code of conduct
+
+We take our open source community seriously and hold ourselves and other contributors to high standards of communication.
+By participating and contributing to this project, you agree to uphold our [Code of conduct](https://www.contributor-covenant.org/version/2/1/code_of_conduct/).
+
+## Report a bug or suggest a feature
+
+1. Search whether your bug or feature was already reported. You can do so on [the GitHub issue tracker of nixos/nix](https://github.com/NixOS/nix/issues).
+
+2. If you were not able to find the bug or feature [open a new issue](https://github.com/NixOS/nix/issues/new/choose)
+
+3. The issue template will guide you in what should be in an issue.
+   The more complete the information, the more likely it can be found by others and the more useful it is in the future.
+   Try to make sure reported bugs can be reproduced by other contributors.
+
+4. Once submitted, do not expect issues to be picked up or solved right away.
+   The only way to ensure this, is to work on the issue yourself. See [Making changes to Nix](#making-changes-to-nix).
+
+## Report a security vulnerability
+
+Check out the [security policy](https://github.com/NixOS/nix/security/policy).
+
+## Making changes to Nix
+
+1. Search for already existing pull requests that might cover the contribution you are about to make.
+   There are many open pull requests that might already do what you intent to work on.
+
+2. Search for related issues that cover what you're going to work on. It could help to mention there that you will work on the issue.
+
+3. Check [the manual](https://nixos.org/manual/nix/stable/contributing/hacking.html) for information on building Nix and running its tests.
+   
+   For contributions on the CLI, please check the manual page on [CLI guidelines](https://nixos.org/manual/nix/stable/contributing/cli-guideline.html).
+   
+4. Make your changes!
+
+5. [Create a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) for your changes
+   * [Mark the pull request as draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) if you're not done with the changes.
+   * Make sure to have [a clean history of commits on your branch by using rebase](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request).
+   * Link related issues in your pull request to inform interested parties and future contributors about your change.
```

Comment:
```suggestion
   * Link related issues in your pull request to inform interested parties and future contributors about your change.
     If your pull request closes one or multiple issues, note that in the description using `Closes: #<number>`, as it will then happen automatically when your change is merged.
```